File: palpite.php
<?php
    // Gerado pelo ChatGPT de acordo com a descrição fornecida por Adilson dos Santos Dantas.
    header("Pragma: no-cache");
    echo "<html>\n";
    echo "<head>\n";
    echo "<title>Palpite para Mega Sena</title>\n";
    echo "<style>\n";
    echo "body { font-family: Arial, Helvetica; }\n";
    echo "</style>\n";
    echo "</head>\n";
    echo "<body>\n";
    echo "<h1>Palpite para Mega Sena</h1>\n";
    echo "<br>\n";
    echo "<p>As dezenas sorteadas são: ";
    $numeros = array();
    for ($i = 0; $i < 6; $i++) {
        $novoNumero = rand(1, 60);
        while (in_array($novoNumero, $numeros)) {
            $novoNumero = rand(1, 60);
        }
        $numeros[$i] = $novoNumero;
        if ($i == 5) {
            echo $numeros[$i];
        } else {
            echo $numeros[$i] . " - ";
        }
    }
    echo "</p>\n";
    echo "<p>As dezenas na ordem são: ";
    sort($numeros);
    for ($i = 0; $i < 6; $i++) {
        if ($i == 5) {
            echo $numeros[$i];
        } else {
            echo $numeros[$i] . " - ";
        }
    }
    echo "</p>\n";
    echo "</body>\n";
    echo "</html>\n";
?>

File: palpite.pas
program MegaSena;
var
  dezenas: array[1..6] of integer;
  i, j: integer;
  num: integer;
  flag: boolean;
begin
  randomize;
  writeln('Palpite para Mega Sena');
  writeln;
  write('As dezenas sorteadas são: ');
  for i := 1 to 6 do
  begin
    repeat
      num := random(60) + 1;
      flag := true;
      for j := 1 to i - 1 do
      begin
        if dezenas[j] = num then
        begin
          flag := false;
          break;
        end;
      end;
    until flag;
    dezenas[i] := num;
    if i < 6 then
    begin
      write(dezenas[i], ' - ');
    end
    else
    begin
      write(dezenas[i]);
    end;
  end;
  writeln;
  write('As dezenas na ordem são: ');
  for i := 1 to 6 do
  begin
    for j := i + 1 to 6 do
    begin
      if dezenas[i] > dezenas[j] then
      begin
        num := dezenas[i];
        dezenas[i] := dezenas[j];
        dezenas[j] := num;
      end;
    end;
    if i < 6 then
    begin
      write(dezenas[i], ' - ');
    end
    else
    begin
      write(dezenas[i]);
    end;
  end;
  writeln;
  writeln;
end.
